### PR TITLE
feat: implement non-standard pagination max_results in Echo service

### PR DESCRIPTION
**Pull request by vchudnov-g**
Includes tests

### CODE REVIEW

File: schema/google/showcase/v1beta1/echo.proto [hunk 1]
```diff
@@ -163,6 +173,20 @@ message PagedExpandRequest {
...
 }
 
+// The request for the PagedExpandLegacy method. New APIs should NOT
+// use this pattern; use PagedExpandRequest instead.
+message PagedExpandLegacyRequest {
+  // The string to expand.
+  string content = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The amount of words to returned in each page.  This is a legacy, non-standard pattern that
+  // violates http://https://google.aip.dev/158
```
Comment (noahdietz) [inline schema/google/showcase/v1beta1/echo.proto:183]: Nit: can you wrap the reasoning/commentary in an aip.dev/not-precedent comment please?

**Review by noahdietz — APPROVED** [review]
LGTM. Will this RPC be "expanded" (no pun intended) to include the map paging once supported? Or will that be a new RPC?

File: schema/google/showcase/v1beta1/echo.proto (hunk 1 above)
Comment (vchudnov-g) [reply schema/google/showcase/v1beta1/echo.proto]: Good idea! I forgot about that.

**Comment by vchudnov-g** [comment]
My plan is to have a separate RPC for paged responses that are maps.

**Pull request merged**